### DISCUSSION TITLE
Handling other genotyping formats?

### DISCUSSION THREAD

**Discussion by smlmbrt** [PGScatalog/pgsc_calc#90] [Ideas]
The current pipeline accepts the following types of genotyping data as input:
- PLINK 1 binary (`.bed/bim/fam`) : https://www.cog-genomics.org/plink/2.0/input#bed
- PLINK 2 binary (`.pgen/pvar/psam`) : https://www.cog-genomics.org/plink/2.0/input#pgen
- Variant call format (`.vcf`) : assuming it's from imputed genotyping data (e.g. the output of Michigan/TopMed imputation server, etc.)

We're wondering if these options are sufficient, or if we're missing other useful inputs? Other possibilities could include:
- BGEN format : https://www.well.ox.ac.uk/~gav/bgen_format/index.html [more compressed than plink]
- Genomic VCF (gVCF) : https://gatk.broadinstitute.org/hc/en-us/articles/360035531812-GVCF-Genomic-Variant-Call-Format [carries information about regions that have been called but are not listed as variant sites/lines (e.g. homozygous reference) which would be necessary for tabulating variant matching statistics] Discussed previously here: https://github.com/PGScatalog/pgsc_calc/issues/50

**Comment by moe1619** [comment]
one vote for gVCF. Helpful for calculating PRS when starting from WGS for example. Thank you!

**Reply by bgulko** [reply]
This is my workflow as well (WGS->gvcf->PGRS). Building a VCF by filtering our gVCF to induce a VCF call from every position mentioned in any PGR catalog model is definitely troublesome. Second vote for gVCF.

**Comment by bnwolford** [comment]
Vote for bgen! As an alternative, I could use a wrapper that creates temporary PLINK files from bgen for just the markers needed for the PGS(s) requested.
